test/Libraries/RevitIntegrationTests/SerializationTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.IO;
+using NUnit.Framework;
+using RevitTestServices;
+using RTF.Framework;
+using Revit.Elements;
+using System.Linq;
+using Autodesk.DesignScript.Geometry;
+using System;
+
+namespace RevitSystemTests
+{
+    [TestFixture]
+    class SerializationTests : RevitSystemTestBase
+    {
+        private string JsonFolderName = "DynamoRevitJsons";
+
+        [Test, TestModel(@".\empty.rfa")]
+        public void SaveXMLTestFilesToJson()
+        {
+            // Get all old XML DYNs
+            var di = new DirectoryInfo(workingDirectory);
+            var fis = new string[] { "*.dyn", "*.dyf" }
+            .SelectMany(i => di.GetFiles(i, SearchOption.AllDirectories))
+            .ToArray();
+            string[] XmlFilePaths = fis.Select(fi => fi.FullName).ToArray();
+
+            // Create new Folder under temp to store Json
+            var tempPath = Path.GetTempPath();
+            var jsonFolder = Path.Combine(tempPath, JsonFolderName);
+
+            if (!System.IO.Directory.Exists(jsonFolder))
+            {
+                System.IO.Directory.CreateDirectory(jsonFolder);
+            }
+
+            foreach (var filePath in XmlFilePaths)
+            {
+                // Open XMLs in new Dynamo Core which should take care of
+                // Migration but element binding might be lost
+                ViewModel.OpenCommand.Execute(filePath);
+                AssertNoDummyNodes();
+
+                // Get new file path under temp folder
+                var fileNameSameStructure = filePath.Split(new string[] { "\\test" }, StringSplitOptions.None).Last();
+                var jsonPath = jsonFolder + fileNameSameStructure;
+                if (File.Exists(jsonPath))
+                {
+                    File.Delete(jsonPath);
+                }
+
+                // Re-save dyn, if folder does not exist, create it
+                FileInfo fileInfo = new FileInfo(jsonPath);
+                if (!fileInfo.Exists)
+                    Directory.CreateDirectory(fileInfo.Directory.FullName);
+
+                ViewModel.SaveAsCommand.Execute(jsonPath);
```

Comment:
does it make sense to verify that the file actually got saved here and some sanity check like it includes a some nodes and node views?

Reply:
Considering this is a single test handling re-save for all DYNs, I would not bail out here. Also did not want to introduce NewtonSoft dependency to D4R

Reply:
it's already there - we need a dep on json.net to add json constructors etc. But no need to check here if it doesn't make sense.